pkg/meta/base.go

```diff
@@ -230,6 +243,86 @@ func (m *baseMeta) checkRoot(inode Ino) Ino {
 	}
 }
 
+func (m *baseMeta) countDirUsage(ctx Context, ino Ino) (space, inodes uint64, err syscall.Errno) {
+	var entries []*Entry
+	if err = m.en.doReaddir(ctx, ino, 1, &entries, -1); err != 0 {
+		return
+	}
+	for _, e := range entries {
+		inodes += 1
+		var newSpace uint64
+		if e.Attr.Typ == TypeFile {
+			newSpace = e.Attr.Length
+		}
+		space += uint64(align4K(newSpace))
+	}
+	return
+}
+
+func (m *baseMeta) GetDirUsage(ctx Context, inode Ino) (space, inodes uint64, err error) {
+	return m.en.doGetDirUsage(ctx, m.checkRoot(inode))
+}
+
+// may cause data race, caller should hold dirUsageLock
+func (m *baseMeta) increDirUsage(ctx Context, ino Ino, space int64, inodes int64) {
+	if inodes == 0 && space == 0 {
+		return
+	}
+	event := m.dirUsageEvents[ino]
+	event.space += space
+	event.inodes += inodes
+	m.dirUsageEvents[ino] = event
+}
+
+func (m *baseMeta) tryIncreDirUsage(ctx Context, ino Ino, space int64, inodes int64) bool {
```

Comment:
we don't need this, increDirUsage does not block

Reply:
Removed